### PR TITLE
SM-276 | feat: dOTC subgraph update to handle ERC 1155

### PR DESCRIPTION
**Pull request by ilamini**
[x] Add ERC1155 Support to the Subgraph

### CODE REVIEW

File: src/mappings/dOTCManager.ts [hunk 1]
```diff
@@ -64,17 +67,9 @@ export function handleNewOrder(event: CreatedOrder): void {
...
-      ? offer.availableAmount.minus(amountPaid)
-      : ZERO_BD
-    offer.save()
-  } else {
-    order.amountPaid = ZERO_BD
-    order.amountToReceive = ZERO_BD
   }
 
   order.orderedBy = event.params.orderedBy
-  order.save()
```
Comment (ElvisKrop) [inline src/mappings/dOTCManager.ts:77]: @ilamini Why did you remove those lines?
At least we need the last line because it is the only place where we make a record of the order

**Review by ElvisKrop — CHANGES_REQUESTED** [review]

File: src/mappings/dOTCManager.ts (hunk 1 above)
Comment (ilamini) [reply src/mappings/dOTCManager.ts:77]: So sorry it was during conflict fixes

File: src/mappings/dOTCManager.ts [hunk 2]
```diff
@@ -64,6 +67,11 @@ export function handleNewOrder(event: CreatedOrder): void {
     )
 
     order.offers = offer.id
+    offer.availableAmount =
+      offer.availableAmount &&
+      offer.availableAmount.minus(
+        bigIntToDecimal(event.params.amountPaid, tokenPaid.decimals),
+      )
```
Comment (ElvisKrop) [inline src/mappings/dOTCManager.ts:74]: @ilamini Those strings are already there, take a look at next lines

**Review by ElvisKrop — APPROVED** [review]
